### ISSUE TITLE
C0303: Trailing Whitespace Error Message Repeats Twice for Space Character After Closing Triple Quotes

### ISSUE THREAD

**Issue by cdreimer-thewriter** [PyCQA/pylint#6936]
### Bug description

A space character after the closing triple quotes for a doc string will cause the ```C0303: Trailing whitespace (trailing-whitespace)``` error message to appear twice in the console output.

board.py:
```python
""" This module has the Board class.
""" 
```

Console output:
```
PS E:\code\chess> pylint board.py
************* Module board
board.py:2:3: C0303: Trailing whitespace (trailing-whitespace)
board.py:2:3: C0303: Trailing whitespace (trailing-whitespace)
```

### Command used

```shell
pylint board.py
```

### Pylint output

```shell
************* Module board
board.py:2:3: C0303: Trailing whitespace (trailing-whitespace)
board.py:2:3: C0303: Trailing whitespace (trailing-whitespace)
```

### Expected behavior

Error message should appear only once in the console output.

### Pylint version

```shell
pylint 2.14.0
astroid 2.11.5
Python 3.10.5 (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 10 / VS Code 1.68.0

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue, I can reproduce that.

**Comment by cgkoutzigiannis** [comment]
Hello! After poking around a bit it seems like there is some kind of issue with the tokens. I run pylint with this code:
```python
"""This is a test.
""" 

class Test:
    pass
```
Here is the list of tokens that gets passed into ```process_tokens```:
```python
[TokenInfo(type=63 (ENCODING), string='utf-8', start=(0, 0), end=(0, 0), line=''),
 TokenInfo(type=3 (STRING), string='"""This is a test.\n"""', start=(1, 0), end=(2, 3), line='"""This is a test.\n""" \n'),
 TokenInfo(type=4 (NEWLINE), string='\n', start=(2, 4), end=(2, 5), line='""" \n'),
 TokenInfo(type=62 (NL), string='\n', start=(3, 0), end=(3, 1), line='\n'),
 TokenInfo(type=1 (NAME), string='class', start=(4, 0), end=(4, 5), line='class Test:\n'),
 TokenInfo(type=1 (NAME), string='Test', start=(4, 6), end=(4, 10), line='class Test:\n'),
 TokenInfo(type=54 (OP), string=':', start=(4, 10), end=(4, 11), line='class Test:\n'),
 TokenInfo(type=4 (NEWLINE), string='\n', start=(4, 11), end=(4, 12), line='class Test:\n'),
 TokenInfo(type=5 (INDENT), string='    ', start=(5, 0), end=(5, 4), line='    pass\n'),
 TokenInfo(type=1 (NAME), string='pass', start=(5, 4), end=(5, 8), line='    pass\n'),
 TokenInfo(type=4 (NEWLINE), string='\n', start=(5, 8), end=(5, 9), line='    pass\n'),
 TokenInfo(type=6 (DEDENT), string='', start=(6, 0), end=(6, 0), line=''),
 TokenInfo(type=0 (ENDMARKER), string='', start=(6, 0), end=(6, 0), line='')]
```
I can see that pylint reads the second line of the comment twice and that's what is causing the double message.

**Comment by Pierre-Sassoulas** [comment]
For future reference the tokenization is done with [tokenize](https://docs.python.org/fr/3/library/tokenize.html) here : https://github.com/PyCQA/pylint/blob/main/pylint/utils/utils.py#L181
